### PR TITLE
Add option for setting selectedIndex in segmentSelect  call-back

### LINKED ISSUE #2229: SelectedItem on segmentedControl does not change after selectedIndex is updated [BUG]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
When changing selectedIndex of a segmentedControl, from the function provided with segmentSelect, nothing happens.

## Describe how to reproduce the bug
1. bind a function to segmentSelect
2. bind a property to selectedIndex
3. change the property bound to selectedIndex to another value than the segment selected from the function bound to segmentSelect

## Which Kirby version was used?
5.3.1

## What was the expected behavior?
The selectedIndex was selected rather than nothing happening

## Add any screenshots

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/stable/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [stable branch](https://github.com/kirbydesign/designsystem/tree/stable) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

### PR DESCRIPTION
**Pull request by mictro**
## Which issue does this PR close?

This PR closes #2229

## What is the new behavior?
Fix to allow for changing selectedIndex in the function bound to segmentSelect. The selectedIndex should be bound to the kirby-segmented-control using a two-way binding '[(segmentSelect)]=....'

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

**Review by jkaltoft — DISMISSED** [review]
image

Looks nice, simple and easy to understand 😁

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts [hunk 1]
```diff
@@ -78,6 +79,8 @@ export class SegmentedControlComponent {
     return this._value;
   }
 
+  @Output() selectedIndexChange = new EventEmitter<number>();
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts:82]: Just out of curiosity, why is this new output needed? 

We should remember to document it in the cookbook too! 😄

Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts:82]: Just out of curiosity, why is this new output needed?

We should remember to document it in the cookbook too! 😄

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts (hunk 1 above)
Comment (mictro) [reply libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts:82]: Will do... - still in progress...
The selected index is both manipulated internally in response to mouse mouse clicks and programmatically. 
Defining this as a one-way binding means it doesn't get updated externally when it is changed internally (get out-of-sync) which in turn means that trying setting this value externally/programmatically afterwards may be ignored.

Comment (mictro) [reply libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts]: Done..

**Review by jkaltoft — APPROVED** [review]

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts [hunk 2]
```diff
@@ -66,13 +66,16 @@ export class SegmentedControlComponent {
...
-      this._selectedIndex = value;
+  @Input() set selectedIndex(index: number) {
+    if (index !== this._selectedIndex) {
+      this._selectedIndex = index;
       this._value = this.items[this.selectedIndex];
+      this.selectedIndexChange.emit(this.selectedIndex);
     }
   }
 
+  @Output() selectedIndexChange = new EventEmitter<number>();
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts:77]: Could you briefly explain why this new output is needed? And it should probably be documented in the API table of the cookbook page too.

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]

**Pull request merged**